### PR TITLE
Feat(store):read transaction from redux store

### PR DESCRIPTION
**Pull request by Tommytrg**
# PR Prelude

Thank you for contributing to sheikah! :)

**Please fill in this template (by putting an `x` inside
the brackets) _before_ filing your PR:**

- [x] I have read and understood [CODE_OF_CONDUCT][code] document.
- [x] I have read and understood [CONTRIBUTING][cont] document.
- [x] I have read and understood [STYLEGUIDE][style] document.
- [x] I have included tests for the changes in my PR. If not, I have included a
  rationale for why I haven't.
- [x] My code is formatted and is accepted by `yarn fmt`.
- [x] **I understand my PR may be closed if it becomes obvious I didn't
  actually perform all of these steps.**

# Why this change is necessary and useful

[Please explain **in detail** why the changes in this PR are needed.]
- Add "RECOMMENDED" literal to options that had it.
- Polish transaction and balance components.
- Read transactions information from redux store.
- Add custom svg's.
- Delete template icons.
- Update color palette.
- Fix font sizes. 

[code]: https://github.com/witnet/sheikah/blob/master/.github/CODE_OF_CONDUCT.md
[cont]: https://github.com/witnet/sheikah/blob/master/.github/CONTRIBUTING.md
[style]: https://github.com/witnet/sheikah/blob/master/docs/STYLEGUIDE.md

### CODE REVIEW

File: app/renderer/ui/components/walletForm/steps/walletEncryptionPassword/style.scss [hunk 1]
```diff
@@ -1,7 +1,11 @@
 :local {
   @import "app/renderer/ui/theme";
+  @import "app/renderer/ui/fonts";
 
   .text {
+    @extend %font14;
```
Comment (aesedepece) [inline app/renderer/ui/components/walletForm/steps/walletEncryptionPassword/style.scss]: Drop this.

File: app/renderer/ui/components/walletForm/steps/walletEncryptionPassword/style.scss [hunk 2]
```diff
@@ -16,6 +20,8 @@
   }
 
   .label {
+    @extend %font14;
```
Comment (aesedepece) [inline app/renderer/ui/components/walletForm/steps/walletEncryptionPassword/style.scss]: Drop this.

File: app/common/runtimeTypes/storage/index.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import * as t from "io-ts"
+
+export const CURRENT_WALLETS_VERSION = 0
+export const CURRENT_WALLET_VERSION = 0
+export const CURRENT_TRANSACTIONS_VERSION = 0
+
+export const Version = t.type({
+  _v: t.number
+}, "Version")
+export type Version = t.TypeOf<typeof Version>
```
Comment (mmartinbar) [inline app/common/runtimeTypes/storage/index.ts]: I'm not sure whether these changes are to be implemented anymore... When talking about the implementation of #401, we decided that transactions are not to be saved in storage for the time being. Keeping that in mind, I think we don't need to version transactions at this point and so, I would remove the definition of CURRENT_TRANSACTIONS_VERSION.

File: app/common/runtimeTypes/storage/transactions.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    // empty if not included in a block
+    blkid: t.string,
+    height: t.number,
+  })
+], "Transaction")
+export type Transaction = t.TypeOf<typeof Transaction>
+
+// Transaction dictionary
+export const Transactions = t.dictionary(t.string, Transaction)
+export type Transactions = t.TypeOf<typeof Transactions>
```
Comment (mmartinbar) [inline app/common/runtimeTypes/storage/transactions.ts]: Same thing here, really :crying_cat_face: . All the types defined in this file come from previous thoughts about #401, so I would remove this file. Just to leave it documented, all the work done about transactions types (this file and related changes) is in [feat-401](https://github.com/mmartinbar/sheikah/tree/feat-401).

File: app/renderer/actions/main.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { actionCreator } from "./helpers"
+import * as Actions from "./actionNames"
+import { Transactions } from "app/common/runtimeTypes/storage/transactions"
+import {
+  SaveTransactionsAction,
+} from "./actionTypes"
+
+// Action creator to save wallet
+export const getTransactions = (transactions: Transactions): SaveTransactionsAction => {
```
Comment (mmartinbar) [inline app/renderer/actions/main.ts]: Perhaps a better name for this function could be `saveTransactions`

File: app/renderer/actions/main.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { actionCreator } from "./helpers"
+import * as Actions from "./actionNames"
+import { Transactions } from "app/common/runtimeTypes/storage/transactions"
+import {
+  SaveTransactionsAction,
+} from "./actionTypes"
+
+// Action creator to save wallet
```
Comment (mmartinbar) [inline app/renderer/actions/main.ts]: Mismatch in comment

File: app/renderer/actions/main.ts (hunk 5 above)
Comment (mmartinbar) [reply app/renderer/actions/main.ts]: And the logic should be updated to the new type of the transactions `ComputedTransactions`

File: app/renderer/store/index.ts [hunk 7]
```diff
@@ -1,10 +1,13 @@
 import { WalletInfos, Wallet } from "app/common/runtimeTypes/storage/wallets"
+import { ComputedTransactions } from "app/renderer/prefilledTransactions"
 
 // Optional state for the wallet
 export type WalletOption = Wallet | false
+export type TransactionsOption = ComputedTransactions
```
Comment (mmartinbar) [inline app/renderer/store/index.ts]: This type definition could be removed, it is not used

File: app/renderer/actions/actionTypes.ts [hunk 8]
```diff
@@ -1,5 +1,8 @@
 import { IActionWithPayload } from "app/renderer/actions/helpers"
 import { Wallet } from "app/common/runtimeTypes/storage/wallets"
+import { ComputedTransaction } from "app/renderer/prefilledTransactions"
 
 // Types for the actions that arrive at the wallet reducer
-export type SaveWalletAction = IActionWithPayload<Wallet>
\ No newline at end of file
+export type SaveWalletAction = IActionWithPayload<Wallet>
+
+export type SaveTransactionsAction = IActionWithPayload<ComputedTransaction>
```
Comment (mmartinbar) [inline app/renderer/actions/actionTypes.ts]: I think it's not a single transaction that should be stored in redux, so it should be `IActionWithPayload<ComputedTransactions>`

File: app/renderer/actions/loginForm.ts [hunk 9]
```diff
@@ -1,11 +1,19 @@
...
 } from "./actionTypes"
 
 // Action creator to save wallet
 export const saveWallet = (wallet: Wallet): SaveWalletAction => {
   return actionCreator<Wallet>(Actions.SAVE_WALLET)(wallet)
+}
+
+// Action creator to save wallet
+export const saveTransactions = (transactions: ComputedTransaction): SaveTransactionsAction => {
+  return actionCreator<ComputedTransaction>(Actions.SAVE_TRANSACTIONS)(transactions)
```
Comment (mmartinbar) [inline app/renderer/actions/loginForm.ts]: Same thing here, it is not a single transaction that needs to be stored in redux, so the parameter should be of type `ComputedTransactions`. Also, the comment needs to be updated.

File: app/renderer/actions/main.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+import { ComputedTransaction } from "app/renderer/prefilledTransactions"
+import * as Actions from "./actionNames"
+import {
+  SaveTransactionsAction,
+} from "./actionTypes"
+
+// Action creator to save wallet
+export const saveTransactions = (transactions: ComputedTransaction): SaveTransactionsAction => {
+  return actionCreator<ComputedTransaction>(Actions.SAVE_TRANSACTIONS)(transactions)
+}
```
Comment (mmartinbar) [inline app/renderer/actions/main.ts]: This whole file can be removed as it just implements a function that is already defined in [here](https://github.com/witnet/sheikah/pull/429/files#diff-21f3a335cf1f229255860adc851fe790)

File: app/renderer/reducers/transactions.ts [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import * as Actions from "app/renderer/actions/actionNames"
+import { SaveTransactionsAction } from "app/renderer/actions/actionTypes"
+import { prefilledTransactions, ComputedTransactions } from "app/renderer/prefilledTransactions"
+
+const defaultTransactrions: ComputedTransactions = prefilledTransactions
```
Comment (mmartinbar) [inline app/renderer/reducers/transactions.ts]: Please, rename to `defaultTransactions`

File: app/renderer/reducers/transactions.ts [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+
+export const transactionsReducer = (state: ComputedTransactions = defaultTransactrions,
+  action: SaveTransactionsAction): ComputedTransactions => {
+  switch (action.type) {
+    case Actions.SAVE_TRANSACTIONS:
+      return prefilledTransactions
+    default:
+      return state
+  }
+}
```
Comment (mmartinbar) [inline app/renderer/reducers/transactions.ts]: Also, I would leave the `prefilledTransactions` out of the reducer (or at least that was the approach taken with the prefilled wallet). The `defaultTransactions` would be an empty array and when handling the `SAVE_TRANSACTIONS` action we just need to return the transactions in the payload of the action itself.

File: app/common/runtimeTypes/storage/index.ts (hunk 3 above)
Comment (Tommytrg) [reply app/common/runtimeTypes/storage/index.ts]: Done :stuck_out_tongue_winking_eye:

File: app/renderer/ui/components/main/sections/wallet/tabs/transactions/index.tsx [hunk 13]
```diff
@@ -36,29 +41,43 @@ class Transactions extends TabComponent<any> {
...
+      />
+    )
+
     return (
       <>
         <div className={styles.left}>
           <Wrapper
             title="PENDING"
-            caption="2 transactions"
+            caption={`${Object.keys(this.props.pendingTransactions).length} transactions`}
```
Comment (mmartinbar) [inline app/renderer/ui/components/main/sections/wallet/tabs/transactions/index.tsx]: Why couldn't this just be the length of the `this.props.pendingTransactions` array?

File: app/renderer/ui/components/main/sections/wallet/tabs/transactions/index.tsx [hunk 14]
```diff
@@ -36,29 +41,43 @@ class Transactions extends TabComponent<any> {
...
             className={styles.pending}
+            empty={!Object.keys(this.props.pendingTransactions).length}
           >
```
Comment (mmartinbar) [inline app/renderer/ui/components/main/sections/wallet/tabs/transactions/index.tsx]: Same thing here

File: app/renderer/ui/components/main/sections/wallet/tabs/transactions/index.tsx [hunk 15]
```diff
@@ -36,29 +41,43 @@ class Transactions extends TabComponent<any> {
...
-            <List
-              dataSource={pendingTransactions}
-              renderItem={PendingTransaction}
-            />
+            {pendingTransactionsList}
           </Wrapper>
           <Wrapper
             title="CONFIRMED"
-            caption="3 transactions"
+            caption={`${Object.keys(this.props.confirmedTransactions).length} transactions`}
```
Comment (mmartinbar) [inline app/renderer/ui/components/main/sections/wallet/tabs/transactions/index.tsx]: Why couldn't this just be the length of the `this.props.confirmedTransactions` array?

File: app/renderer/ui/components/main/sections/wallet/tabs/transactions/index.tsx [hunk 16]
```diff
@@ -36,29 +41,43 @@ class Transactions extends TabComponent<any> {
...
             actions={confirmedOptions}
             className={styles.confirmed}
+            empty={!Object.keys(this.props.confirmedTransactions).length}
```
Comment (mmartinbar) [inline app/renderer/ui/components/main/sections/wallet/tabs/transactions/index.tsx]: Same thing here

File: app/renderer/selectors/index.tsx [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+        {
+          status: transaction.timelock === 0 ? "timelocked" : "UNCONFIRMED",
+          address: `${transaction.addresses[0]}`,
+          receiver: transaction.type === "incoming",
+          amount: `${transaction.value}`,
+          vestingTime: transaction.timelock === 0 ? `Vesting on ${transaction.date}` : ""
+        }
+      ))
+      : []
+}
```
[the remark this answers is not in the document]
Comment (Tommytrg) [reply app/renderer/selectors/index.tsx]: This should be change when we add a reselect function to get that information as computed properties.

File: app/renderer/ui/components/transaction/confirmed/index.tsx [hunk 18]
```diff
@@ -46,7 +46,7 @@ export default class ConfirmedTransaction extends React.Component<ConfirmedTrans
         </p>
         <div>
           <p className={styles.date}>{this.props.date}</p>
-          <p className={styles.block}>Confirmed in block #{this.props.block}</p>
+          <p className={styles.block}>Confirmed in epoch #{this.props.block}</p>
```
Comment (mmartinbar) [inline app/renderer/ui/components/transaction/confirmed/index.tsx]: I would leave the `block` text. @aesedepece told us that block height and epoch are the same thing and in the sketch it appears as `block`.

File: app/renderer/ui/components/walletForm/steps/walletSeedTypeSelection/index.tsx [hunk 19]
```diff
@@ -47,6 +48,7 @@ export default class WalletSeedTypeSelection extends React.Component<Iprops> {
             <ButtonOption
               recommended={true}
               onClick={this.props.nextStep.newPrefilledSeed}
+              secondaryText="RECOMMENDED"
```
Comment (mmartinbar) [inline app/renderer/ui/components/walletForm/steps/walletSeedTypeSelection/index.tsx]: :clap:

File: app/renderer/ui/containers/Main.tsx [hunk 20]
```diff
@@ -65,6 +74,8 @@ const mapDispatchToProps = (dispatch: Dispatch<IAction>): DispatchProps => {
  */
 const mapStateToProps = (state: StoreState): StateProps => {
   return ({
+    confirmedTransactions: filterConfirmedTransactions(state.transactions),
+    pendingTransactions: filterPendingTransactions(state.transactions)
```
Comment (mmartinbar) [inline app/renderer/ui/containers/Main.tsx]: :heart:

**Pull request merged**